Voc det 1

vassilinast opened the pull request:
1) Code cleaning 
2) Finalized vocalization detection method. In audio_process.py we added a new function prepare_features where spectral energy and some other features are calculated, in order to be used in vocalization detection. 
Basically, we calculate the total spectral energy, the mean spectral energy of an area around max freq, and max energy for each
 time frame, in order to be used for vocalization detection.
The new vocalization detection method is more robust to background noise.
We also use a post filtering method to discard a few false postiives.

vassilinast commented:
https://docs.google.com/spreadsheets/d/1Kv2thpGLPzoflB7GHtn0pt0cJJ-utLC2u1QLK98DKP0/edit?usp=sharing

tyiannak commented:
could u please also provide either in a comment in this PR or as an update in the readme file (preferably), a detailed step-by-step instruction list of how to reproduce the reported evaluation results ? (start from getting the data etc)

tyiannak commented on audio_recognize.py:
```diff
@@ -57,37 +57,29 @@ def cluster_syllables(syllables, specgram, sp_freq,
-            # plt.imshow(temp_image.T)
-            # plt.show()
+        vec=[np.mean(temp_image),np.var(temp_image), np.mean(cur_image-np.amax(cur_image)), np.var(cur_image-np.amax(cur_image))]
+        if (vec[0]-0.7*kmeans_centers[1,0]>=0 and vec[1]-0.7*kmeans_centers[1,2]>=0):
```
could you explain this rule here and why is this repeated in main_live as well? maybe we should make it a function?

tyiannak commented on main_live.py:
```diff
@@ -56,15 +58,35 @@ def signal_handler(signal, frame):
+        temp_image = cur_image/np.amax(cur_image)
+
+        vec = [np.mean(temp_image),np.var(temp_image), np.mean(cur_image-np.amax(cur_image)), np.var(cur_image-np.amax(cur_image))]
+        if (vec[0]-0.7*kmeans_centers[1,0]>=0 and vec[1]-0.7*kmeans_centers[1,2]>=0):
```
same here

vassilinast commented:
I removed post processing code. I also added syllables_comp.py which can be used for comparing detected vocalizations between two csv files. This can be used to reproduce results to compare to ground truth files. I will update the README about how to use it.
